flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/ColumnarArray.java

```diff
@@ -0,0 +1,332 @@
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setFloat(int ordinal, float value) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setDouble(int ordinal, double value) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setDecimal(int i, Decimal value, int precision) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setTimestamp(int ordinal, SqlTimestamp value, int precision) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNotNullAt(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullLong(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullInt(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullBoolean(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullByte(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullShort(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullFloat(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public void setNullDouble(int pos) {
+		throw new UnsupportedOperationException("Not support the operation!");
+	}
+
+	@Override
+	public boolean[] toBooleanArray() {
+		boolean[] res = new boolean[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getBoolean(i);
+		}
+		return res;
+	}
+
+	@Override
+	public byte[] toByteArray() {
+		byte[] res = new byte[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getByte(i);
+		}
+		return res;
+	}
+
+	@Override
+	public short[] toShortArray() {
+		short[] res = new short[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getShort(i);
+		}
+		return res;
+	}
+
+	@Override
+	public int[] toIntArray() {
+		int[] res = new int[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getInt(i);
+		}
+		return res;
+	}
+
+	@Override
+	public long[] toLongArray() {
+		long[] res = new long[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getLong(i);
+		}
+		return res;
+	}
+
+	@Override
+	public float[] toFloatArray() {
+		float[] res = new float[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getFloat(i);
+		}
+		return res;
+	}
+
+	@Override
+	public double[] toDoubleArray() {
+		double[] res = new double[length];
+		for (int i = 0; i < length; i++) {
+			res[i] = getDouble(i);
+		}
+		return res;
+	}
+
+	public BaseArray copy() {
```

Comment:
Can we move this method to serializer or somewhere?
So we don't need this elementType field.